### PR TITLE
chore: support location ID in mail destinations

### PR DESCRIPTION
**Pull request by mr-flannery**
~Closes~ contributes to SAP/cloud-sdk-backlog#1126.

requires extending the e2e tests

### CODE REVIEW

File: packages/mail-client/src/socket-proxy.ts [hunk 1]
```diff
@@ -18,8 +27,9 @@ export async function customAuthRequestHandler(jwt: string): Promise<Buffer> {
     // The actual value of the JWT in its encoded form
     ...jwtBytes,
     // Length of the Cloud Connector location ID (0 if no Cloud Connector location ID is used)
-    0x00
+    0x00,
```
Comment (marikaner) [inline packages/mail-client/src/socket-proxy.ts:30]: [q] Shouldn't the length also be adjusted?

Comment (mr-flannery) [reply packages/mail-client/src/socket-proxy.ts:30]: oh, duh, yeah, it probably should 🤦 thanks

File: packages/mail-client/src/socket-proxy.ts [hunk 2]
```diff
@@ -5,11 +5,26 @@
...
   const jwtBytesLength = jwtBytes.length;
+
   const jwtLengthBuffer = Buffer.alloc(4);
   jwtLengthBuffer.writeUInt32BE(jwtBytesLength, 0);
+
+  const cloudConnectorLocationIdBytes = cloudConnectorLocationId
+    ? [...Buffer.from(cloudConnectorLocationId)]
+    : [];
+
+  const cloudConnectorLocationIdLengthBuffer = Buffer.alloc(1);
```
Comment (marikaner) [inline packages/mail-client/src/socket-proxy.ts:22]: [q] Is it intended that this is different than the `jwtLengthBuffer` - I would expect the same format is needed and that the functionality can be extracted into a separate function then. Something with a signature similar to this.

```ts
function createBuffer(): { buffer; length; }
```

**Review by marikaner — APPROVED** [review]
👍

**Pull request merged**